client/my-sites/site-settings/form-discussion.jsx

```diff
@@ -70,332 +78,292 @@ module.exports = React.createClass( {
+									valueLink={ this.linkState( 'thread_comments_depth' ) }
+									disabled={ this.state.fetchingSettings }
+									onClick={ this.recordClickEventAndStop.bind( this, 'Selected Comment Nesting Level' ) }>
+										<option value="2">2</option>
+										<option value="3">3</option>
+										<option value="4">4</option>
+										<option value="5">5</option>
+										<option value="6">6</option>
+										<option value="7">7</option>
+										<option value="8">8</option>
+										<option value="9">9</option>
+										<option value="10">10</option>
+								</FormSelect>
 							}
 						} )
 						}</span>
-				</label>
-				<label>
-					<input
-					name="page_comments"
-					type="checkbox"
-					id="page_comments"
-					checkedLink={ this.linkState( 'page_comments' ) }
-					disabled={ this.state.fetchingSettings }
-					onClick={ this.recordEvent.bind( this, 'Clicked Break Comments Into Pages Checkbox' ) }
-					/>
+				</FormLabel>
+				<FormLabel>
+					<FormCheckbox
+						name="page_comments"
+						id="page_comments"
+						checkedLink={ this.linkState( 'page_comments' ) }
+						disabled={ this.state.fetchingSettings }
+						onClick={ this.recordEvent.bind( this, 'Clicked Break Comments Into Pages Checkbox' ) } />
 					<span>{
 						this.translate( 'Break comments into pages with {{numComments /}} top level comments per page and the {{firstOrLast /}} page displayed by default', {
 							components: {
-								numComments: <input
-								name="comments_per_page"
-								type="number"
-								step="1"
-								min="0"
-								id="comments_per_page"
-								valueLink={ this.linkState( 'comments_per_page' ) }
-								className="small-text"
-								disabled={ this.state.fetchingSettings }
-								onClick={ this.recordEvent.bind( this, 'Clicked Comments Per Page Field' ) }
-								onKeyPress={ this.recordEventOnce.bind( this, 'typedCommentsPerPage', 'Typed in Comments Per Page Field' ) }
-								/>,
-								firstOrLast: <select
-								className="is-compact"
-								name="default_comments_page"
-								valueLink={ this.linkState( 'default_comments_page' ) }
-								disabled={ this.state.fetchingSettings }
-								onClick={ this.recordClickEventAndStop.bind( this, 'Selected Comment Page Display Default' ) }
-								>
-									<option value="newest">{ this.translate( 'last' ) }</option>
-									<option value="oldest">{ this.translate( 'first' ) }</option>
-								</select>
+								numComments: <FormTextInput
+									name="comments_per_page"
+									type="number"
+									step="1"
+									min="0"
+									id="comments_per_page"
+									valueLink={ this.linkState( 'comments_per_page' ) }
+									className="small-text"
+									disabled={ this.state.fetchingSettings }
+									onClick={ this.recordEvent.bind( this, 'Clicked Comments Per Page Field' ) }
+									onKeyPress={ this.recordEventOnce.bind( this, 'typedCommentsPerPage', 'Typed in Comments Per Page Field' ) } />,
```

Comment:
Maybe another earmark for a future PR, but these items are awfully big and mainly static. It could be clearer if they were wrapped into a specialized component with fewer props:

```js
const { linkState, recordEvent, recordEventOnce } = this;
const { fetchingSettings } = this.state;

numComments: <CommentCountInput {...{ disabled: fetchingSettings, linkState, recordEvent, recordEventOnce } } />
```

Since React auto-binds the functions, these should retain their `this` in `<CommentCountInput />`

```js
CommentCountInput = ( { disabled, linkState, recordEvent, recordEventOnce } ) => (
	<FormTextInput
		name="comments_per_page"
		type="number"
		step="1"
		min="0"
		id="comments_per_page"
		valueLink={ linkState( 'comments_per_page' ) }
		className="small-text"
		disabled={ disabled }
		onClick={ () => recordEvent( 'Clicked Comments Per Page Field' ) }
		onKeyPress={ () => recordEventOnce( 'typedCommentsPerPage', 'Typed in Comments Per Page Field' ) } />
);
```

Note that I'm simply using a new shorthand for [stateless React components](http://facebook.github.io/react/docs/reusable-components.html#stateless-functions) here. It's a way of defining simple and "pure" components that results in some speedups starting in React 0.14 I think.

There's a chance that wouldn't work due to the way `linkState` functions. If that's the case a normal React component with `render()` would be the alternate choice.

I like this solution because the functionality can be modularized and the noise of these particulars can be more easily hidden _vis-a-vis_ less distracting.